lnbits/wallets/corelightningrest.py

```diff
@@ -0,0 +1,241 @@
+import asyncio
+import json
+import random
+from typing import AsyncGenerator, Dict, Optional
+
+import httpx
+from loguru import logger
+
+from lnbits import bolt11 as lnbits_bolt11
+from lnbits.settings import settings
+
+from ..core.models import Payment, PaymentStatus
+from .base import InvoiceResponse, PaymentResponse, StatusResponse, Unsupported, Wallet
+from .macaroon import load_macaroon
+
+
+class CoreLightningRestWallet(Wallet):
+    def __init__(self):
+        macaroon = settings.corelightning_rest_macaroon
+        assert macaroon, "missing cln-rest macaroon"
+
+        self.macaroon = load_macaroon(macaroon)
+
+        url = settings.corelightning_rest_url
+        if not url or not macaroon:
+            raise Exception("cannot initialize corelightning-rest")
+
+        self.url = url[:-1] if url.endswith("/") else url
+        self.auth = {
+            "macaroon": self.macaroon,
+            "encodingtype": "hex",
+            "accept": "application/json",
+        }
+
+        self.cert = settings.corelightning_rest_cert or False
+        self.client = httpx.AsyncClient(verify=self.cert, headers=self.auth)
+        self.last_pay_index = 0
+        self.statuses = {
+            "paid": True,
+            "complete": True,
+            "failed": False,
+            "pending": None,
+        }
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        r = await self.client.get(f"{self.url}/v1/channel/localremotebal", timeout=5)
+
+        if r.is_error or "error" in r.json():
+            try:
+                data = r.json()
+                error_message = data["error"]
```

Comment:
- it is possible for `data` not to have an  `"error"` field (`or` condition at line `54`)